olp-cpp-sdk-core/tests/thread/ContinuationTest.cpp

```diff
@@ -53,145 +56,167 @@ class ContinuationTest : public testing::Test {
+  ASSERT_EQ(result.GetError().GetHttpStatusCode(),
             static_cast<int>(olp::http::ErrorCode::CANCELLED_ERROR));
 }
 
 TEST_F(ContinuationTest, FinallyNotSet) {
-  auto cont =
+  auto continuation =
       Create([](olp::thread::ExecutionContext, std::function<void(int)>) {});
 
-  cont.Run();
+  continuation.Run();
 }
 
 TEST_F(ContinuationTest, CancelWithFinallyNotSet) {
-  auto cont =
+  auto continuation =
       Create([](olp::thread::ExecutionContext, std::function<void(int)>) {});
 
-  cont.CancelToken().Cancel();
-  cont.Run();
+  continuation.CancelToken().Cancel();
+  continuation.Run();
 }
 
 TEST_F(ContinuationTest, Failed) {
-  std::promise<olp::client::ApiError> api_error_promise;
+  std::promise<ResponseType<int>> api_error_promise;
   auto future = api_error_promise.get_future();
 
-  auto cont = Create([](olp::thread::ExecutionContext context,
-                        std::function<void(int)>) {
-                context.SetError(olp::client::ApiError::NetworkConnection());
-              }).Finally([&api_error_promise](Response<int> response) {
-    api_error_promise.set_value(response.GetError());
-  });
+  auto continuation =
+      Create([](olp::thread::ExecutionContext context,
+                std::function<void(int)>) {
+        context.SetError(olp::client::ApiError::NetworkConnection());
+      }).Finally([&api_error_promise](ResponseType<int> response) {
+        api_error_promise.set_value(std::move(response));
+      });
+
+  continuation.Run();
 
-  cont.Run();
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
   auto result = future.get();
 
-  EXPECT_EQ(result.GetErrorCode(), olp::client::ErrorCode::NetworkConnection);
+  ASSERT_EQ(result.GetError().GetErrorCode(),
+            olp::client::ErrorCode::NetworkConnection);
 }
 
 TEST_F(ContinuationTest, NoCrashAfterCallingMethodsAfterRun) {
-  std::promise<olp::client::ApiError> api_error_promise;
+  std::promise<ResponseType<int>> api_error_promise;
   auto future = api_error_promise.get_future();
 
-  auto cont = Create([](olp::thread::ExecutionContext context,
-                        std::function<void(int)>) {
-                context.SetError(olp::client::ApiError::NetworkConnection());
-              }).Finally([&api_error_promise](Response<int> response) {
-    api_error_promise.set_value(response.GetError());
-  });
+  auto continuation =
+      Create([](olp::thread::ExecutionContext context,
+                std::function<void(int)>) {
+        context.SetError(olp::client::ApiError::NetworkConnection());
+      }).Finally([&api_error_promise](ResponseType<int> response) {
+        api_error_promise.set_value(std::move(response));
+      });
 
-  cont.Run();
+  continuation.Run();
+
+  ASSERT_EQ(future.wait_for(kMaxWaitMs), std::future_status::ready);
   auto result = future.get();
 
-  EXPECT_EQ(result.GetErrorCode(), olp::client::ErrorCode::NetworkConnection);
+  ASSERT_EQ(result.GetError().GetErrorCode(),
```

Comment:
You also need to check if the response is not successful before getting the error.

Reply:
done